### ISSUE TITLE
order/class-member-order: false positives when an inner class body's first line is a comment

### ISSUE THREAD

**Issue by Shilo** [atelico/gdstyle#5]
### Minimal reproduction

```gdscript
class_name Foo extends Object
## doc.
class Bar extends RefCounted:
	## doc for a.
	var a: int
```

Trigger: an inner class whose **first body line is a comment on its own line** (`#` or `##`). Removing that leading comment — or moving it **inline** onto the declaration — makes the warning disappear:

```gdscript
class_name Foo extends Object
## doc.
class Bar extends RefCounted:
	var a: int ## inline doc — clean, no warning
	var b: int
```

### Command you ran

```
gdstyle check repro.gd
```

### What gdstyle did

```
repro.gd
  5:2 warning variable declaration should appear before inner class (see GDScript style guide for ordering) [order/class-member-order]
      	var a: int
      	^

1 file checked, 1 warning found.
```

Every member of the inner class is flagged as `order/class-member-order` ("should appear before inner class"), not just the first.

### What you expected

No diagnostics. `var a` is the first member *inside* `Bar`, correctly placed — not an out-of-order member of `Foo`.

### gdstyle version

`gdstyle 0.1.5` (latest). Also reproduces on `0.1.4`. The v0.1.5 patch only addressed an unrelated `@abstract` formatter issue (#4) and does not touch this path. Verified against the official `gdstyle-x86_64-pc-windows-msvc.zip` release binaries.

### Platform

Windows (x86_64) — but this is platform-independent (parser/lexer logic).

### Additional context

**Root cause** (traced through `src/lexer.rs` and `src/parser.rs`):

1. `lexer.rs::process_indentation` does **not** emit an `Indent` token for a comment-only line — when it sees the indented comment right after `class Bar:` (while the indent stack top is still the outer level), it preserves the stack and defers. The `Indent` is emitted only at the next *real* line (`var a`). So the token stream is `… Colon, Newline, DocComment, Newline, Indent, Var, …`.
2. `parser.rs::parse_indented_block` calls `skip_newlines()` (which skips `Newline` but **not** `Comment`/`DocComment`), then returns an **empty body** if the next token isn't `Indent`. It hits the `DocComment` first → returns `[]`.
3. With an empty inner-class body, the `var`/`func` tokens fall through to the **outer** parse loop and become siblings *after* the `InnerClass` node. The `order/class-member-order` rule (itself correct, and recurses into inner classes properly) then flags them as appearing "before inner class".

This also explains why the **inline** form is clean: an inline trailing comment rides on the same code line that carries the `Indent`, so `parse_indented_block` sees `Indent` immediately and parses the body normally.

**Suggested fix:** in `parse_indented_block`, skip (ideally retain, so they stay available to the doc-comment-attachment logic) leading `Comment`/`DocComment` tokens before testing for `Indent` — or have the lexer emit the `Indent` ahead of a leading comment line that is more-indented than the current block.

**Test gap:** `tests/fixtures/bad_ordering.gd` has no inner classes, so this path is untested. A fixture with a comment-first inner-class body (plus a `correct` counterpart) would lock in the fix. The existing `ordering.rs` unit tests build the AST directly, bypassing the parser, so they can't catch this — a parser-level or fixture-level test is needed.

**Comment by Shilo** [comment]
Confirmed still present in **v0.1.5** (tested the official `gdstyle-x86_64-pc-windows-msvc.zip` release binary; `gdstyle 0.1.5`). The v0.1.5 patch addressed a different parser issue (#4, `@abstract` formatter explosion) and does not touch this code path.

Repro under v0.1.5:
```gdscript
class_name Foo extends Object
## doc.
class Bar extends RefCounted:
	## doc for a.
	var a: int
```
```
4:2 warning variable declaration should appear before inner class [order/class-member-order]
1 file checked, 1 warning found.
```

One extra data point that corroborates the root cause: the trigger is a comment on its **own line** as the first body element. An **inline** trailing doc comment on the first body line does **not** trigger it:
```gdscript
class_name Foo extends Object
## doc.
class Bar extends RefCounted:
	var a: int ## inline doc.   # clean — no warning
	var b: int
```
That's consistent with the mechanism in the original report: a standalone leading comment is emitted as a `DocComment`/`Comment` token *before* the `Indent`, so `parse_indented_block` bails with an empty body; an inline comment rides on the same code line that carries the `Indent`, so the block parses normally.

**Comment by w4nderlust** [comment]
Fixed in v0.1.6 . Your trace through `process_indentation` was exactly right: the deeper-indent comment case was falling into the same-indent branch, so no `Indent` got emitted and the body fell through to the outer scope. Split into three cases by relative indent now, with deeper falling through to `Indent` emission. Added one lexer test and two integration tests (your repro plus a stacked `#`/`##` variant). Thanks for the detailed write-up, and for catching the inner-class fixture gap.

**Comment by Shilo** [comment]
> Fixed in v0.1.6 .

Great! Thank you for the quick response and fix. This library has been quite useful. Have a good day!

**Comment by w4nderlust** [comment]
> Great! Thank you for the quick response and fix. This library has been quite useful. Have a good day!

Glad you are liking it and that it's being useful! Let me now if you encounter other issues.
And spread the word :)
